sfge/src/main/java/com/salesforce/rules/DmlUtil.java

```diff
@@ -0,0 +1,93 @@
+package com.salesforce.rules;
+
+import com.salesforce.graph.EnumUtil;
+import java.util.Optional;
+import java.util.TreeMap;
+
+public class DmlUtil {
```

Comment:
1. It looks like this class is only used by the new rule? If so, then you could move this class into the `dmlinloop` package, possibly adding an `operations` sub-package.
2. Util classes should all be declared as `final`.

Reply:
If, in the future, we need an `operations` package, we can add it.